exist-core/src/main/java/org/exist/xquery/value/Type.java

```diff
@@ -579,20 +579,20 @@ public static int getCommonSuperType(final int type1, final int type2) {
         //Don't introduce a shortcut (starting at getSuperType(type1) here
         //type2 might be a super-type of type1
         int t;
-        for (t = type1; t != ITEM; t = getSuperType(t)) {
+        for (t = type1; t != ITEM && t != ANY_TYPE; t = getSuperType(t)) {
```

Comment:
Would that be one less check for the more frequent case t being ITEM?
`!(t==ITEM || t==ANY_TYPE)`